openstack/blockstorage/v2/volumes/results.go

```diff
@@ -57,6 +77,25 @@ type Volume struct {
 	Multiattach bool `json:"multiattach"`
 }
 
+func (r *Volume) UnmarshalJSON(b []byte) error {
+	type tmp Volume
+	var s struct {
+		tmp
+		CreatedAt gophercloud.JSONRFC3339MilliNoZ `json:"created_at"`
```

Comment:
Ah I see.  If we use a time.Time here, we will be unable to process dates that do not contain timezone information, which are also valid ISO8601 dates.  However, the current code won't handle those dates properly either.  It won't return an error when Unmarshalling but the dates will almost always be interpreted incorrectly, unless both client and server are in the UK or Ireland and it's deep midwinter.  ISO8601 dates without timezone information are in local time but time.Parse interprets such date strings as being in UTC.  I think the following code would do the correct thing in all cases:

```
	t, err := time.Parse(time.RFC3339, s)
	if err != nil {
		t, err = time.ParseInLocation("2006-01-02T15:04:05", s, time.Local)
	}
```

Perhaps the best thing to do would be to create a new goperhcloud time type that uses the above code in its UnmarshalText method.